core/system.h

```diff
@@ -17,41 +17,56 @@ class PluginImplBase;
 class System
 {
 public:
+    /**
+     * @brief Constructor.
+     * @param system_id MAVLink system id.
+     * @param comp_id MAVLink component id.
```

Comment:
To avoid using `MAVLinkSystem` terminology, I think we've to provide `SystemImpl` may be in future. Isn't it ?

Reply:
I don't know how that question relates to this section of code :-0

Reply:
I meant that we'd to mention about MAVLink in DroneCore's public header file `system.h`. :)
Anyway app developer won't be able to use this constructor, so I was thinking is it necessary to mention about it.

Reply:
It appears in the docs because it is a public API, so best to say something ...

Perhaps in the class overview doc we should add:

> System objects are not created directly by application code, but are returned by the DroneCore class. 

In the  constructor we could have:
```
 /**
 * @brief Constructor.
 *
 * This constructor is not (and should not be) directly called by application code.
 * 
 * @param system_id System id.
 * @param comp_id Component id.
*/
```

Thoughts?